api_v3/services/LiveStreamService.php

```diff
@@ -577,4 +577,102 @@ public function regenerateStreamTokenAction($entryId)
 
 	}
 
+    /**
+     * Archive a live entry which was recorded
+     *
+     * @action archive
+     * @param string $liveEntryId
+     * @param KalturaLiveEntryArchiveOptions $archiveOptions
+     * @return bool
+     * @throws KalturaAPIException
+     * @throws KalturaClientException
+     * @throws PropelException
+     */
+	public function archiveAction($liveEntryId, KalturaLiveEntryArchiveOptions $archiveOptions)
+    {
+        $liveEntry = entryPeer::retrieveByPK($liveEntryId);
+        /** @var LiveStreamEntry $liveEntry */
+        if (!$liveEntry)
+        {
+            KalturaLog::err("Live entry id [" . $liveEntryId . "] not found");
+            throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $liveEntryId);
+        }
+        $cuePoints = CuePointPeer::retrieveByEntryId($liveEntryId);
```

Comment:
This could return large result so better do in parts or even considering moving this entire logic to batch .